### PR TITLE
Add User Item Repository and Fix Jacoco and CI

### PR DESCRIPTION
**Pull request by omarelmalki**
Fixed Jacoco and CI to recognize tests in test/ folder.
Adapted Item class to have String IDs for more compatibilty.
Added User Item Repository (feel free to uncomment and adapt/remove them in your code @pmcintyr when you finalize the repo).

### CODE REVIEW

File: app/src/main/java/com/github/brugapp/brug/data/UserItemRepository.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+//                } else {
+//                    Log.d(TAG, "No such document")
+//                }
+//            }
+//            .addOnFailureListener { exception ->
+//                Log.d(TAG, "get failed with ", exception)
+//            }
+//        return item
+//    }
+}
```
Comment (Mouniraki) [inline app/src/main/java/com/github/brugapp/brug/data/UserItemRepository.kt:98]: Why did you comment the whole class content ? Will it be deleted in a future commit ?

**Review by Mouniraki — CHANGES_REQUESTED** [review]
Good job, mainly refactorings as I see but consider my comment related on the UserItemRepository class. Also, since I see that you worked with the Firebase API, did you have issues to correctly retrieve data and create the User object, since everything is asynchronous ?

File: app/src/main/java/com/github/brugapp/brug/data/UserItemRepository.kt (hunk 1 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/data/UserItemRepository.kt:98]: The content of the class conflicts with what @pmcintyr is currently working on as I needed it during my work. By commenting it, I leave the choice of keeping/adapting the code to him like we have discussed together after the last scrum meeting.

**Review by hamzahass — APPROVED** [review]
Good job on fixing the issue related to Unit tests not being recognized by Jacoco. However I have two questions : 
- How is making the item id a string better for compatibility ?
- But also (as Mounir mentioned) why did you comment all the UserItemRepository class ?

**Review by Mouniraki — APPROVED** [review]
Approving the PR from your explanations :)

**Comment by omarelmalki** [comment]
> Good job on fixing the issue related to Unit tests not being recognized by Jacoco. However I have two questions :
> 
> * How is making the item id a string better for compatibility ?
> * But also (as Mounir mentioned) why did you comment all the UserItemRepository class ?

* In our database (Firebase) item ids are strings.
* I commented it to leave @pmcintyr the control over what is available in public in the repositories to avoid duplicate functions. We have already discussed that together.

**Pull request merged**